api/v1alpha1/envoyproxy_types.go

```diff
@@ -73,6 +73,15 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	MergeGateways *bool `json:"mergeGateways,omitempty"`
+
+	// OverloadManager defines the configuration for the Overload Manager in envoy.
+	// Visit https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/overload_manager/overload_manager#config-overload-manager
+	// to learn more about the syntax.
+	// Overload manager is disabled by default.
+	// After the feature is enabled, following two overload actions are configured to Envoy:
+	// 1.Shrink heap action is executed when 95% of the maximum heap size is reached.
+	// 2.Envoy will stop accepting requests when 98% of the maximum heap size is reached.
+	OverloadManager *OverloadManager `json:"overloadManager,omitempty"`
```

Comment:
imo we shouldnt expose envoyisms such as `overloadManager` directly, but do it in a way that is relevant to the EG user

Reply:
You're right，the `MaxHeapSizeBytes` value should be exposed directly to enable the overload manager. I'm still thinking about how to set this value specifically.